### PR TITLE
fix: v9 DatePicker compat popup is not hidden from the a11y tree

### LINKED ISSUE #31818: [Bug]: v9 DatePicker compat popup is not accessible in latest Chromium
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
N/A
```

### Are you reporting Accessibility issue?

yes

### Reproduction

repros on the docs site

## Actual Behavior
The DatePicker popup is not in the accessibility tree (does not repro on v8, related to v9 being modal). This is caused by a leftover `aria-owns` attribute that moves the popup from the end of the DOM to within the hidden tree. This was necessary when the datepicker was non-modal, but is breaking access now that it is modal.

The practical impact is that all screen readers and voice control are unable to access the dialog on the latest version of Chromium and Safari.

## Expected Behavior
The DatePicker is accessible 😅

### Logs

_No response_

### Requested priority

High

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by smhigley**
## Previous Behavior

A stray `aria-owns` from before DatePicker was modal caused the popup to be an a11y child of the hidden tree, which made it inaccessible to screen reader users and voice control users.

Note: this is not an issue in v8, since the DatePicker there is non-modal.

## New Behavior

Removes the `aria-owns` and fixes the breaking a11y bug. Also makes a few incremental improvements, including:

- Removes the unnecessary extra group role between the dialog and the table.
- Moves the selected date & today's date string to the table's accname
- Moves `aria-hidden` for placeholder rows to the `<tr>` element from the cells, preventing empty rows from showing in the a11y tree
- Removes unnecessary `aria-hidden` from the span inside date buttons

## Related Issue(s)
- Fixes #31818

### CODE REVIEW

**Review by khmakoto — APPROVED** [review]

File: packages/react-components/react-calendar-compat/library/src/components/Calendar/Calendar.tsx [hunk 1]
```diff
@@ -340,19 +340,13 @@ export const Calendar: React.FunctionComponent<CalendarProps> = React.forwardRef
...
-        className={classes.root}
-        onKeyDown={onDatePickerPopupKeyDown}
-      >
+      <div id={id} ref={forwardedRef} className={classes.root} onKeyDown={onDatePickerPopupKeyDown}>
         <div className={classes.liveRegion} aria-live="polite" aria-atomic="true">
           <span>{selectedDateString}</span>
         </div>
         {isDayPickerVisible && (
           <CalendarDay
+            ariaLabel={selectionAndTodayString}
```
Comment (khmakoto) [inline packages/react-components/react-calendar-compat/library/src/components/Calendar/Calendar.tsx:349]: Seems like CI is complaining about this, maybe it is `aria-label`?

```suggestion
            aria-label={selectionAndTodayString}
```

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-calendar-compat/library/src/components/Calendar/Calendar.tsx (hunk 1 above)
Comment (sopranopillow) [reply packages/react-components/react-calendar-compat/library/src/components/Calendar/Calendar.tsx:349]: yup, we can only use kebab cased aria attributes

File: packages/react-components/react-calendar-compat/library/src/components/CalendarDay/CalendarDay.tsx [hunk 2]
```diff
@@ -75,6 +76,7 @@ export const CalendarDay: React.FunctionComponent<CalendarDayProps> = props => {
       </div>
       <CalendarDayGrid
         {...propsWithoutStyles}
+        gridLabel={`${monthAndYear}, ${ariaLabel}`}
         componentRef={dayGrid}
```
Comment (sopranopillow) [inline packages/react-components/react-calendar-compat/library/src/components/CalendarDay/CalendarDay.tsx:80]: same here
```suggestion
      <CalendarDayGrid
        {...propsWithoutStyles}
        gridLabel={`${monthAndYear}, ${props['aria-label']}`}
        componentRef={dayGrid}
```

File: packages/react-components/react-calendar-compat/library/src/components/CalendarDay/CalendarDay.tsx [hunk 3]
```diff
@@ -26,6 +26,7 @@ export const CalendarDay: React.FunctionComponent<CalendarDayProps> = props => {
 
   const {
     strings,
+    ariaLabel,
```
Comment (sopranopillow) [inline packages/react-components/react-calendar-compat/library/src/components/CalendarDay/CalendarDay.tsx:29]:
```suggestion
```

File: packages/react-components/react-calendar-compat/library/src/components/CalendarDayGrid/CalendarGridRow.tsx [hunk 4]
```diff
@@ -50,7 +51,7 @@ export const CalendarGridRow: React.FunctionComponent<CalendarGridRowProps> = pr
     : '';
 
   return (
-    <tr role={ariaRole} className={rowClassName} key={weekIndex + '_' + week[0].key}>
+    <tr role={ariaRole} aria-hidden={ariaHidden} className={rowClassName} key={weekIndex + '_' + week[0].key}>
```
Comment (sopranopillow) [inline packages/react-components/react-calendar-compat/library/src/components/CalendarDayGrid/CalendarGridRow.tsx:54]:
```suggestion
    <tr role={ariaRole} aria-hidden={props['aria-hidden']} className={rowClassName} key={weekIndex + '_' + week[0].key}>
```

File: packages/react-components/react-calendar-compat/library/src/components/CalendarDayGrid/CalendarGridRow.tsx [hunk 5]
```diff
@@ -29,6 +29,7 @@ export interface CalendarGridRowProps extends CalendarDayGridProps {
  */
 export const CalendarGridRow: React.FunctionComponent<CalendarGridRowProps> = props => {
   const {
+    ariaHidden,
```
Comment (sopranopillow) [inline packages/react-components/react-calendar-compat/library/src/components/CalendarDayGrid/CalendarGridRow.tsx:32]:
```suggestion
```

File: packages/react-components/react-datepicker-compat/library/src/components/DatePicker/useDatePicker.tsx [hunk 6]
```diff
@@ -364,7 +364,6 @@ export const useDatePicker_unstable = (props: DatePickerProps, ref: React.Ref<HT
 
   const inputRoot = slot.always(props.root, {
     defaultProps: {
-      'aria-owns': open ? popupSurfaceId : undefined,
```
Comment (sopranopillow) [inline packages/react-components/react-datepicker-compat/library/src/components/DatePicker/useDatePicker.tsx:367]: just confirming that this was intended :)

**Review by sopranopillow — APPROVED** [review]

File: packages/react-components/react-datepicker-compat/library/src/components/DatePicker/useDatePicker.tsx (hunk 6 above)
Comment (smhigley) [reply packages/react-components/react-datepicker-compat/library/src/components/DatePicker/useDatePicker.tsx:367]: yup! This is the fix for the OH-GOD-FIRE bug in the original issue 😅

This prop made sense when the dialog was non-modal (it needed to be inline in the a11y tree), but now that it's modal, it moves the dialog to the hidden a11y tree, making it inaccessible :P

**Pull request merged**